sfge/src/main/java/com/salesforce/rules/Violation.java

```diff
@@ -302,6 +303,33 @@ public int compareTo(Violation otherViolation) {
         }
     }
 
+    public static final class LimitReachedViolation extends Violation {
+
+        /**
+         * @param message - The value to be inserted into the violation's {@link #message}
+         *     attribute.
+         * @param vertex
+         */
+        public LimitReachedViolation(String message, SFVertex vertex) {
+            super(
+                    String.format(
+                            UserFacingMessages.RuleViolationTemplates
+                                    .LIMIT_REACHED_VIOLATION_MESSAGE,
+                            message),
+                    vertex);
+
+            // TODO: reconsider the name and category. We want users to take followup action.
+            this.ruleName = "LimitReached";
+            this.category = INTERNAL_ERROR_CATEGORY;
+            this.description = "";
+            this.severity = AbstractRule.SEVERITY.LOW.code;
+            // NOTE: For now, we're hardcoding the URL to the URL for the Apex FLS violation rule,
+            // since that's the only rule that can throw this violation. This may change in a future
+            // release.
+            this.url = ApexFlsViolationRule.URL;
```

Comment:
@teresa-allen-sfdc Once we have a separate section in our docs on how to handle this situation, we can update the link here.